### ISSUE TITLE
Not working. npm install -g strongloop

### ISSUE THREAD

**Issue by shaununderwood** [strongloop-community/express-example-modular#3]
I've been around the houses attempting to install strongloop on a new box without any success. Work fine on my mac but a new oceandigital ubuntu 14.04 server, vanilla or with node preinstalled fails to install strongloop. Also tried with nvm and various node and iojs versions. installed build essentials. no joy. Really frustrating.

nvm reports : Now using io.js v2.2.1 (npm v2.11.0) (same as mac setup but also tried with v0.10.36 and v0.10.38, and no joy).

Command: sudo npm install -g strongloop
Also tried: npm install -g strongloop
Also tried npm install strongloop

Every time the process just bottoms out! Sometimes a failed or killed note will appear, other time it just stops dead, as in the below (which is output from npm install strongloop)

Output from process

```
> heapdump@0.3.6 install /home/goman-root/node_modules/strongloop/node_modules/strong-supervisor/node_modules/heapdump
> node-gyp rebuild

make: Entering directory `/home/goman-root/node_modules/strongloop/node_modules/strong-supervisor/node_modules/heapdump/build'
  CXX(target) Release/obj.target/addon/src/heapdump.o
  SOLINK_MODULE(target) Release/obj.target/addon.node
  SOLINK_MODULE(target) Release/obj.target/addon.node: Finished
  COPY Release/addon.node
make: Leaving directory `/home/goman-root/node_modules/strongloop/node_modules/strong-supervisor/node_modules/heapdump/build'

> strong-fork-syslog@1.2.3 install /home/goman-root/node_modules/strongloop/node_modules/strong-supervisor/node_modules/strong-fork-syslog
> node-gyp rebuild

make: Entering directory `/home/goman-root/node_modules/strongloop/node_modules/strong-supervisor/node_modules/strong-fork-syslog/build'
  CXX(target) Release/obj.target/syslog/syslog.o
  SOLINK_MODULE(target) Release/obj.target/syslog.node
  SOLINK_MODULE(target) Release/obj.target/syslog.node: Finished
  COPY Release/syslog.node
make: Leaving directory `/home/goman-root/node_modules/strongloop/node_modules/strong-supervisor/node_modules/strong-fork-syslog/build'

> strong-agent@1.6.0 install /home/goman-root/node_modules/strongloop/node_modules/strong-agent
> node-gyp rebuild || exit 0

make: Entering directory `/home/goman-root/node_modules/strongloop/node_modules/strong-agent/build'
  CXX(target) Release/obj.target/strong-agent/src/strong-agent.o
In file included from ../src/strong-agent.cc:15:0:
../src/watchdog.h: In function ‘void strongloop::agent::watchdog::PatchSyscall()’:
../src/watchdog.h:170:52: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
   *reinterpret_cast<volatile uint64_t*>(address + 0) = 0xF8EBFAEBFCEBFEEBULL;
                                                    ^
../src/watchdog.h:199:52: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
   *reinterpret_cast<volatile uint32_t*>(address + 0) =  // movabs $0, %rax
                                                    ^
../src/watchdog.h:200:75: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
       0xB848 | (0xFFFF0000 & *reinterpret_cast<volatile uint32_t*>(address));
                                                                           ^
\
Killed
goman-root@gomantest2:~$ 
goman-root@gomantest2:~$   SOLINK_MODULE(target) Release/obj.target/strong-agent.node
  SOLINK_MODULE(target) Release/obj.target/strong-agent.node: Finished
  COPY Release/strong-agent.node
make: Leaving directory `/home/goman-root/node_modules/strongloop/node_modules/strong-agent/build'
```

Any advice or remedy to this?

**Comment by rmg** [comment]
@shaununderwood I have a hunch you are running out of memory, because none of the Digital Ocean images enable swap by default. Would I be correct in guessing you were using a 512mb instance?

Here's a quick fix:

``` sh
$ sudo dd if=/dev/zero of=/swap bs=1M count=1024
$ sudo mkswap /swap
$ sudo swapon /swap
```

To make that permanent you'll need to add a line to your `/etc/fstab`, something like:

```
/swap  swap         swap    defaults   0       0
```

Does that get you up and running?

**Comment by shaununderwood** [comment]
@rmg Thanks for coming back to me.

Your hunch was correct, a 512mb instance. And I tried your settings and SUCCESS! It worked! So, I just updated the two instances I have and they now compile without such symptoms.

On the same note, earlier in the day I found the discussion below which actually worked for me as well, but that may very well have been due to remnant of prior install attempts masking the otherwise blatant failure. 

http://stackoverflow.com/questions/28524942/issues-installing-strongloop-loopback-in-ubuntu-14-04-lts

Well, at best then thats two ways to overcome the same symptom.

Thanks rmg!
